Zero balances in wallet

# Bug Report

## Summary
For **10 minutes**, all account balances displayed as **0**.

## Expected Behavior
Balances should be correctly displayed at all times.

## Actual Behavior
- Account balances were shown as **0** for 10 minutes.
- Issue occurred **twice in the past two days**.

### Screenshot
![Image](https://github.com/user-attachments/assets/87aa792b-7609-47ac-a272-7d43b1964b2f)

## Steps to Reproduce
- No clear reproduction steps.
- Occurred **twice** over the past **two days**.

## Additional Information

- **App Version:** Release 2.33  
- **OS:** Android, iOS  
- **Logs:** [Download Logs](https://drive.google.com/file/d/1jrWsJd2JqZgB8Tp1YLNTDPXjw6wp9SQI/view?usp=sharing)  

### Relevant Log Entry
```
2025-02-27T14:47:31.630Z	ERROR	market/market.go:106	Error fetching prices
{“error”: “cryptocompare.error: hystrix: timeout, coingecko.error: hystrix: timeout, cryptocompare-proxy.error: Get ***8: context deadline exceeded (Client.Timeout exceeded while awaiting headers)”}
```

```
2025-02-27T14:47:06.721Z ERROR    unknown         market/market.go:236 Error fetching prices
    error: 'cryptocompare.error: hystrix: timeout, coingecko.error: context deadline exceeded
      (Client.Timeout or context cancellation while reading body), cryptocompare-proxy.error:
      Get "https://prod.api.status.im/cryptocompare//data/all/coinlist": context deadline
      exceeded (Client.Timeout exceeded while awaiting headers)'

```


<img width="709" alt="Image" src="https://github.com/user-attachments/assets/a46e986d-ef75-4d53-b45a-a7293b53ba52" />

```
	cb := circuitbreaker.NewCircuitBreaker(circuitbreaker.Config{
		Timeout:               10000,
		MaxConcurrentRequests: 100,
		SleepWindow:           300000,
		ErrorPercentThreshold: 25,
	})
```

So the http request will fail if the speed is less than 16Mbps

Updated: http.Transport automatically adds gzip encoding by default, in this case it's 3.5Mb 
With 5sec timeout it's 5600kbps speed required

--- a/services/wallet/market/market.go
+++ b/services/wallet/market/market.go
@@ -53,7 +53,7 @@ type Manager struct {
 
 func NewManager(providers []thirdparty.MarketDataProvider, feed *event.Feed) *Manager {
 	cb := circuitbreaker.NewCircuitBreaker(circuitbreaker.Config{
-		Timeout:               10000,
+		Timeout:               60000,
 		MaxConcurrentRequests: 100,
 		SleepWindow:           300000,
 		ErrorPercentThreshold: 25,

--- a/services/wallet/thirdparty/http_client.go
+++ b/services/wallet/thirdparty/http_client.go
@@ -5,6 +5,7 @@ import (
 	"context"
 	"encoding/json"
 	"io/ioutil"
+	"net"
 	"net/http"
 	netUrl "net/url"
 	"time"
@@ -19,7 +20,8 @@ type BasicCreds struct {
 }
 
 type HTTPClient struct {
-	client *http.Client
+	client     *http.Client
+	maxRetries int
 }
 
 func NewHTTPClient() *HTTPClient {
@@ -27,6 +29,34 @@ func NewHTTPClient() *HTTPClient {
 		client: &http.Client{
 			Timeout: requestTimeout,
 		},
+		maxRetries: maxNumOfRequestRetries,
+	}
+}
+
+// NewHTTPClientWithDetailedTimeouts creates a new HTTPClient with separate timeouts for
+// connection establishment and data transfer
+func NewHTTPClientWithDetailedTimeouts(
+	dialTimeout time.Duration,
+	tlsHandshakeTimeout time.Duration,
+	responseHeaderTimeout time.Duration,
+	requestTimeout time.Duration,
+	maxRetries int,
+) *HTTPClient {
+	transport := &http.Transport{
+		DialContext: (&net.Dialer{
+			Timeout: dialTimeout, // Timeout for establishing a connection
+		}).DialContext,
+		TLSHandshakeTimeout:   tlsHandshakeTimeout,   // Timeout for TLS handshake
+		ResponseHeaderTimeout: responseHeaderTimeout, // Timeout for receiving response headers
+		IdleConnTimeout:       90 * time.Second,      // How long to keep idle connections
+	}
+
+	return &HTTPClient{
+		client: &http.Client{
+			Transport: transport,
+			Timeout:   requestTimeout, // Overall request timeout
+		},
+		maxRetries: maxRetries,
 	}
 }
 
@@ -45,9 +75,14 @@ func (c *HTTPClient) DoGetRequest(ctx context.Context, url string, params netUrl
 	}
 
 	var resp *http.Response
-	for i := 0; i < maxNumOfRequestRetries; i++ {
+	maxRetries := c.maxRetries
+	if maxRetries <= 0 {
+		maxRetries = maxNumOfRequestRetries // Use default if not set
+	}
+
+	for i := 0; i < maxRetries; i++ {
 		resp, err = c.client.Do(req)
-		if err == nil || i == maxNumOfRequestRetries-1 {
+		if err == nil || i == maxRetries-1 {
 			break
 		}
 		time.Sleep(200 * time.Millisecond)

--- a/services/wallet/thirdparty/market/coingecko/client.go
+++ b/services/wallet/thirdparty/market/coingecko/client.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"strings"
 	"sync"
+	"time"
 
 	"golang.org/x/exp/maps"
 
@@ -64,8 +65,21 @@ type Client struct {
 }
 
 func NewClient() *Client {
+	// Configure HTTP client with detailed timeouts:
+	// - 5 seconds for connection establishment (dialTimeout)
+	// - 5 seconds for TLS handshake (tlsHandshakeTimeout)
+	// - 10 seconds for receiving response headers (responseHeaderTimeout)
+	// - 30 seconds for overall request timeout (requestTimeout)
+	httpClient := thirdparty.NewHTTPClientWithDetailedTimeouts(
+		5*time.Second,  // dialTimeout
+		5*time.Second,  // tlsHandshakeTimeout
+		5*time.Second,  // responseHeaderTimeout
+		20*time.Second, // requestTimeout
+		5,              // retries
+	)
+
 	return &Client{
-		httpClient: thirdparty.NewHTTPClient(),
+		httpClient: httpClient,
 		tokens:     make(map[string][]GeckoToken),
 		baseURL:    baseURL,
 	}

--- a/services/wallet/thirdparty/market/cryptocompare/client.go
+++ b/services/wallet/thirdparty/market/cryptocompare/client.go
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"net/url"
 	"strings"
+	"time"
 
 	"github.com/status-im/status-go/services/wallet/thirdparty"
 	"github.com/status-im/status-go/services/wallet/thirdparty/utils"
@@ -64,9 +65,18 @@ func NewClientWithParams(params Params) *Client {
 		}
 	}
 
+	// Configure HTTP client with detailed timeouts:
+	httpClient := thirdparty.NewHTTPClientWithDetailedTimeouts(
+		5*time.Second,  // dialTimeout
+		5*time.Second,  // tlsHandshakeTimeout
+		5*time.Second,  // responseHeaderTimeout
+		20*time.Second, // requestTimeout
+		5,              // retries
+	)
+
 	return &Client{
 		id:         params.ID,
-		httpClient: thirdparty.NewHTTPClient(),
+		httpClient: httpClient,
 		baseURL:    params.URL,
 		creds:      creds,
 	}

